### DISCUSSION TITLE
Composite only add one image

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#5747] [Help] [answered]
Hi,

I want to add two images side by side with a gap between them, in another one, the output.
I don't want to use gravity because I need a gap between each pictures.

So I calculated that an increasing offset of 300 is good.
 
The following code

```
Magick::Geometry global_geom;
global_geom.height( 1024 );
global_geom.width( 1024 );
Magick::Image output_img( global_geom, Magick::Color( 0, 0, 0, 0 ) );

uint32_t offset = 100;

for ( auto &path_image_to_composite_to_output_img : image_list_paths ){

        Magick::Geometry geom_el;
        geom_el.width( 200 );
        geom_el.height( global_geom.height() );

        Magick::Image local_image( path_image_to_composite_to_output_img );
        local_image.resize( geom_el );

        output_img.composite( local_image, offset, 0 );

        offset += 300;

}

output_img.write(my/path);
```

Will only composite the first image at the proper offset, but not the second one (it doesn't appear).
For the test, the second image is the same as the first, to reduce errors.

Adjusted sizes for the two images to composite have been double checked and we know it fits in a 1024x1024 image (each image is 200 pixels large).

How may I solve this?

**Comment by ghost** [comment]
Ok, changing the output_img backgroun dcolor to red, shows that the second image was there, but black.
How so?

**Comment by snibgo** [comment]
What is the declaration of the `composite` function that you are using?

**Comment by ghost** [comment, marked as answer]
So the issue was the type of the CompositeOperator.

I switched to Magick::OverCompositeOp and everything went as expected!
